modules/mono/mono_gd/gd_mono_class.cpp

```diff
@@ -464,9 +464,18 @@ const Vector<GDMonoClass *> &GDMonoClass::get_all_delegates() {
 		return delegates_list;
 	}
 
+	// If the class is generic we must use the generic type definition.
+	MonoClass *klass = mono_class;
+	MonoReflectionType *reftype = mono_type_get_object(mono_domain_get(), get_mono_type());
+	if (GDMonoUtils::Marshal::type_is_generic(reftype)) {
```

Comment:
I don't think you need this helper function. You should be able to do the same with `mono_type_get_type(this->get_mono_type()) == MONO_TYPE_GENERICINST`, but I haven't tested.

Reply:
It works, so I changed it.